src/coreclr/pal/src/misc/sysinfo.cpp

```diff
@@ -613,9 +613,16 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
     {
         int64_t cacheSizeFromSysctl = 0;
         size_t sz = sizeof(cacheSizeFromSysctl);
-        const bool success = sysctlbyname("hw.l3cachesize", &cacheSizeFromSysctl, &sz, nullptr, 0) == 0
+        const bool success = false
+            // macOS-arm64: Since macOS 12.0, Apple added ".perflevelX." to determinate cache sizes for efficiency 
+            // and performance cores separetely. "perflevel0" stands for "performance"
```

Comment:
should these be included only under `#if defined(TARGET_OSX)` ?

Reply:
Makes sense, wrapped

Reply:
All of these are actually macOS specific. FreeBSD is the only other OS that supports `sysctlbyname`, but it doesn't have support for any of the names we use here. But I don't think it is a big deal whether we filter them out here or not.

Reply:
@janvorli are you suggesting to remove if-defs for better readability?

Reply:
I don't have a strong preference here, I just wanted to clarify it. I am fine with leaving your PR as-is or removing the ifdef. Whatever you prefer.